src/js/media-captions-list.js

```diff
@@ -0,0 +1,133 @@
+import MediaChromeListbox from './media-chrome-listbox.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import { Window as window, Document as document } from './utils/server-safe-globals.js';
+import { MediaUIAttributes, MediaStateReceiverAttributes, MediaUIEvents } from './constants.js';
+import { nouns } from './labels/labels.js';
+import { parseTextTracksStr, parseTextTrackStr, formatTextTrackObj } from './utils/captions.js';
+
+
+class MediaCaptionsList extends MediaChromeListbox {
+  #subs = [];
+  #caps = [];
+
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
+      MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+      MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
```

Comment:
if the media-chrome captions and subtitles and selected list was modeled differently, it would be a lot easier to work with.

Reply:
Upvote. @heff I believe we chatted about this. Tl;dr - If we had attributes and corresponding events that treated captions and subtitles as a "single list" (would need to add the type to our string-serialized representation), this would significantly reduce complexities.

Reply:
Agree with consolidating. The only tradeoff I remember was that we were trying to be technically accurate with the terms "subtitles" / "captions" and not conflate two the terms.

Practically speaking, this did cause confusion when building the player Demuxed 2022 theme because I think:

- The test streams they were using had subtitles, so the theme worked in testing
- The real live stream on the day of the event had captions, and the theme didn't account for that

Also, I didn't fully confirm this but I think we had a bug where in Chrome it was being treated as captions and in Safari it was treated as subtitles, so that was hella confusing.

Reply:
> I think we had a bug where in Chrome it was being treated as captions and in Safari it was treated as subtitles

That's because Safari treats [any characteristic on subtitles tracks](https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-4-4-6-1:~:text=A%20SUBTITLES%20Rendition%20MAY%20include%20the%20following%20characteristics) in the manifest as a caption, while hls.js doesn't treat only treats both as a caption? Also, I think this may have been updated since.

> technically accurate with the terms "subtitles" / "captions" and not conflate two the terms

Basically, only the US treats them materially different. Most of the rest of the world treats them both as subtitles, but captions as "Subtitles for the Deaf and Hard of Hearing" (SDH). Most UIs show them in a single list and then specify that a particular track is specifically a captions track rather than just a regular subtitles track.

> this did cause confusion when building the player Demuxed 2022 theme

Another reason to just have a single list.

Reply:
Let's do it. 👍 (in a follow up)